File: app/src/main/java/com/example/theduckcardchatsystem/ui/model/DiffUtilCalback.java
package com.example.theduckcardchatsystem.ui.model;

import androidx.recyclerview.widget.DiffUtil;

import java.util.List;

public class DiffUtilCalback extends DiffUtil.Callback {
    private List<CommonModel> oldList;
    private List<CommonModel> newList;

    public DiffUtilCalback(List<CommonModel> oldList, List<CommonModel> newList) {
        this.oldList = oldList;
        this.newList = newList;
    }


    @Override
    public int getOldListSize() {
        return oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {

        return oldList.get(oldItemPosition).getTimeStamp() == newList.get(newItemPosition).getTimeStamp();
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition) == newList.get(newItemPosition);
    }
}

File: app/src/main/java/com/example/theduckcardchatsystem/ui/model/DiffUtilCalback.kt
package com.example.theduckcardchatsystem.ui.model

import com.example.theduckcardchatsystem.ui.model.CommonModel.timeStamp
import com.example.theduckcardchatsystem.ui.model.CommonModel
import androidx.recyclerview.widget.DiffUtil

class DiffUtilCalback(
    private val oldList: List<CommonModel>,
    private val newList: List<CommonModel>
) : DiffUtil.Callback() {
    override fun getOldListSize(): Int {
        return oldList.size
    }

    override fun getNewListSize(): Int {
        return newList.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition].timeStamp === newList[newItemPosition].timeStamp
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition] == newList[newItemPosition]
    }
}